deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvStages.java

```diff
@@ -103,6 +103,17 @@ public void update(@Context SecurityContext sc,
         if (origBean.getStage_type() != EnvType.DEFAULT && origBean.getStage_type() != environBean.getStage_type()) {
             throw new TeletaanInternalException(Response.Status.BAD_REQUEST, "Modification of stage type is not allowed!");
         }
+        // TODO: isUserInRole takes a string and does not seem implemented... it should take the value from Role enum
+        if(environBean.getSOX() && !sc.isUserInRole("SYSTEMADMIN")) {  // TODO: see note above
```

Comment:
This isn't working, need to get a better idea of how to check in this situation.

Reply:
how about call authorizer with required role "SYSTEMADMIN" in case request contains SOX field change.

Reply:
Calling authorizer with required role "SYSTEMADMIN" looks clean if possible.

Reply:
2 to check:
POST /v1/envs/{envName}/user_roles
POST /v1/envs/{envName}/group_role